adagrams/game.py

```diff
@@ -1,11 +1,163 @@
+     #LP's version (with coworking!)
 
 def score_word(word):
-    pass
+    letter_scores = make_score_dict()
+    score = get_score(word.upper(), letter_scores)
+    return score
+    
+#challenge: compare words to an imported dictionary to see if it's a word!
+#I would do this, but I don't know how to install packages here. I don't think it'll work in this window.
+
+def make_score_dict():
+    #set up dictionary of score values
+    #we want the form letter are keys and scores are the values
+
+    letter_scores = {}
+    point1 = (1, ["A", "E", "I", 'O', 'U', 'L', 'N', 'R', 'S', 'T'])
+    point2 = (2, ['D', 'G'])
+    point3 = (3, ['B', 'C', 'M', 'P']) 
+    point4 = (4, ['F', 'H', 'V', 'W', 'Y'])
+    point5 = (5, ['K']) 
+    point8 = (8, ['J', 'X'])
+    point10 = (10, ['Q', 'Z'])
+    point_info = [point1, point2, point3, point4, point5, point8, point10]
+    
+    for point_group in point_info:
+        for letter in point_group[1]:
+            letter_scores[letter] = point_group[0]
+            
+    return letter_scores
+    
+def get_score(word, letter_scores):
+    #loop through the word, adding up the value associated with the letter (key) of the dict!
+    score = 0
+    for char in word:
+        #look for the character in the keys of letter_scores.
+        #add the associated value to score
+        score += letter_scores[char]
+        #print(char, word, score)
+    if len(word) >= 7:
+        score += 8
+            
+    return score
+
+# def score_word(word): #Abby
+#     score_values = []
+#     score_dict = { 1: ["A","E", "I", "O", "U", "L", "N", "R", "S", "T"], 2: ["D", "G"], 3: ["B", "C", "M", "P"], 4: ["F","H", "V", "W", "Y"], 5: ["K"], 8: ["J", "X"], 10: ["Q", "Z"]}
+#     word = word.upper()
+
+#     if len(word) >= 7: 
+#         score_values.append(8)
+#     for character in word:
+#         for letter_value, letters in score_dict.items():
+#             for letter in letters:
+#                 if letter == character: 
+#                     score_values.append(letter_value)
+    
+#     return sum(score_values)
+
+#-------------end wave 3---------
+
+
+#------ Begin Wave 4------------
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    This gets the word with the highest score. 
+    This will be a list of tuples (word, score) of type (str, int)
+    """
+    words_scores_list = []
+    
+    for word in word_list: 
+        words_scores_list.append((word, score_word(word)))
+    sorted_words_scores_list = sorted(words_scores_list, key=itemgetter(1), reverse=True)
+
+    max_score = sorted_words_scores_list[0][1]
+    potential_winners = []
+    for word, score in sorted_words_scores_list:
+        if score == max_score:
+            potential_winners.append((word, score, len(word)))
+    
+    if len(potential_winners) == 1:
+        return potential_winners[0]
+
+    sorted_winners = sorted(potential_winners, key = itemgetter(2),reverse=True)
```

Comment:
Very cool way to sort by the words' length! Another approach without having `len(word)` as another tuple value is to use a more custom lambda function key, something like:
`key=lambda potential_winner: len(potential_winner[0])`